### PR TITLE
Adding the ability to create a set of dummy data based on a schema

### PR DESCRIPTION
**Pull request by serroba**
## Overview
This PR introduces enhancements to the TSSchema data structure generation logic, specifically within the `GenerateDummyData` method. Additionally, it expands our test coverage to include scenarios both with and without predefined metric values.

## Changes

- **GenerateDummyData Method**: Adjusted to ensure it correctly generates the desired data structure, allowing for more accurate simulation of Timestream data insertion scenarios.

- **Test Suite Expansion**: Added a new test case to cover the scenario where no predefined values are provided. This test verifies the structural integrity of the generated data and checks that generated values fall within an expected range.

-**Schema Definition Enhancement**: The Schema type has been expanded to support a mapping from table names to measure names, which then map to a record containing both dimensions and metric names. This enhancement allows for a more detailed definition of the data model, accommodating the complexities of real-world datasets.

-**Introduction of Dimensions**: The schema now supports dimensions for each measure, allowing for a richer and more detailed data representation. This addition is crucial for scenarios where the dimensional context of data points is essential for accurate simulation and analysis.

-**Dummy Data Generation**: We've added a GenerateDummyData method that creates mock data entries based on the schema's structure, including both metric names and dimensions. This functionality is invaluable for testing and development, allowing for the creation of realistic datasets without the need for actual data ingestion.

## Purpose
These changes aim to improve the flexibility of TSSchema's data generation capabilities.

## Code Example

Bellow, we see how this changes provide enhance functionality to the library:

```go
package main

import (
	"fmt"
	"time"

	timestream "github.com/EvergenEnergy/TimeSchema/pkg"
)

func main() {
	// Define a schema with dimensions and metric names
	schema := timestream.Schema[string, string]{
		"WeatherStation": { // Table name
			"EnvironmentalData": { // Measure name
				Dimensions: []string{"LocationID", "SensorType"},
				MetricNames: []string{"Temperature", "Humidity"},
			},
		},
	}

	// Initialize a new TSSchema instance with the defined schema
	tsSchema := timestream.NewTSSchema(schema)

	// Define predefined values for the metrics
	predefinedValues := timestream.PredefinedValues[string]{
		"Temperature": 22.5, // Assuming a temperature of 22.5 degrees Celsius
		"Humidity":    60.0, // Assuming a humidity of 60%
	}

	// Generate dummy data using the schema and predefined values
	now := time.Now() // Timestamp for the data
	dbName := "EnvironmentalDataDB"
	dummyData := tsSchema.GenerateDummyData(dbName, now, predefinedValues)

	// Output the generated dummy data
	for _, writeInput := range dummyData {
		fmt.Printf("DatabaseName: %s\n", *writeInput.DatabaseName)
		fmt.Printf("TableName: %s\n", *writeInput.TableName)
		for _, record := range writeInput.Records {
			fmt.Printf("Time: %s\n", *record.Time)
			fmt.Printf("MeasureName: %s\n", *record.MeasureName)
			fmt.Println("Dimensions:")
			for _, dimension := range record.Dimensions {
				fmt.Printf(" - %s: %s\n", *dimension.Name, *dimension.Value)
			}
			fmt.Println("MeasureValues:")
			for _, measureValue := range record.MeasureValues {
				fmt.Printf(" - %s (%s): %s\n", *measureValue.Name, measureValue.Type, *measureValue.Value)
			}
		}
		fmt.Println("----------")
	}
}
```

### CODE REVIEW

File: pkg/schema.go [hunk 1]
```diff
@@ -61,10 +73,81 @@ func (s TSSchema[T]) GetMeasureNameFor(metricName T) (string, error) {
...
 	if !ok {
 		return v.tableName, fmt.Errorf("metric name %T not found", metricName)
 	}
 	return v.tableName, nil
 }
+
+type PredefinedValues[T comparable] map[T]float64
+
+// GenerateDummyData generates dummy data based on the schema structure.
+// This function now uses the telemetry.MetricName type for keys.
```
Comment (ch-evergen) [inline pkg/schema.go:87]: I don't think `telemetry.MetricName` is a valid type here right?

**Comment by ch-evergen** [comment]
Looks really handy! So if you wanted to write a row of nulls with a given schema, would you simply pass a map of empty values to `GenerateDummyData` ?

It would be great for this very informative PR description to be added to the README :)

**Comment by serroba** [comment]
Good point. Will update the readme

File: pkg/schema.go (hunk 1 above)
Comment (serroba) [reply pkg/schema.go:87]: correct, this needs to go

**Comment by serroba** [comment]
> So if you wanted to write a row of nulls with a given schema, would you simply pass a map of empty values to GenerateDummyData ?

I am not sure that would actually work, as we are saying that the type must be float. I wasn't thinking to provide `null` values. We probably can add support to that though if needed. ATM, only accepts floats, NOT NULL.

**Pull request merged**